renderer/components/Activity/Payment/Payment.js

```diff
@@ -1,18 +1,34 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { FormattedMessage, FormattedTime, injectIntl, intlShape } from 'react-intl'
+import { decodePayReq } from '@zap/utils/crypto'
 import { Box, Flex } from 'rebass'
 import { Message, Text } from 'components/UI'
 import { CryptoValue, FiatValue } from 'containers/UI'
 import messages from './messages'
 
 const Payment = ({ payment, showActivityModal, nodes, currencyName, intl }) => {
-  const displayNodeName = pubkey => {
-    const node = nodes.find(n => pubkey === n.pub_key)
-    if (node && node.alias.length) {
-      return node.alias
+  const displayNodeName = payment => {
+    // First try to get the pubkey from payment.path
+    let pubkey = payment.path && payment.path[payment.path.length - 1]
+
+    // If we don't have a pubkey, try to get it from the payment request.
+    if (!pubkey && payment.payment_request) {
+      const paymentRequest = decodePayReq(payment.payment_request)
+      pubkey = paymentRequest.payeeNodeKey
+    }
+
+    // If we know the pubkey, try to lookup its alias.
+    if (pubkey) {
+      const node = nodes.find(n => pubkey === n.pub_key)
+      if (node && node.alias.length) {
+        return node.alias
+      }
+      return pubkey.substring(0, 10)
```

Comment:
should we use `truncateNodePubkey` from `network.js` ?

Reply:
Good call. I've extracted the `truncateNodePubkey` method into a standalone utility method and then updated the codebase to use this.